MeetupPlatformApi/Controllers/AuthenticationController.cs

```diff
@@ -84,4 +150,26 @@ public async Task<IActionResult> GetCurrentUserInfo()
         var outputDto = mapper.Map<UserOutputDto>(user);
         return Ok(outputDto);
     }
+
+    [HttpPut("me/credentials")]
+    [Authorize]
+    public async Task<IActionResult> ChangeCredentials([FromBody] UserCredentialsChangeDto credentialsChangeDto)
+    {
+        var currentUserInfo = authenticationManager.GetCurrentUserInfo(User);
+        var user = await context.Users.SingleOrDefaultAsync(user => user.Id == currentUserInfo.UserId);
+        var usernameAlreadyTaken = await context.Users.AnyAsync(user => user.Username == credentialsChangeDto.Username);
```

Comment:
What will happen, if we provide new password, but old username?

Reply:
This username already taken. We've discussed it and come to the conclusion that username changing in any events.

Reply:
It looks like there's been a misunderstanding.
The user should be able to keep their current name when changing the password. Just like vice versa - he should be able to keep his old password, but change the name.
The current implementation allows you to leave the old password, but does not allow you to leave the old username. This will need to be corrected.